b/collect/index.php

```diff
@@ -45,32 +45,33 @@
 ob_end_clean();
 
 $response = array();
+
 //Agent communication using REST protocol
 if (isset($_GET['action'])) {
+   $pfAgent        = new PluginFusioninventoryAgent();
+   $pfTaskjobstate = new PluginFusioninventoryTaskjobstate();
+   $pfTaskjoblog   = new PluginFusioninventoryTaskjoblog();
+   $pfCollect      = new PluginFusioninventoryCollect();
+
    switch ($_GET['action']) {
 
       case 'getJobs':
          if(isset($_GET['machineid'])) {
-            $pfAgent        = new PluginFusioninventoryAgent();
-            $pfTaskjobstate = new PluginFusioninventoryTaskjobstate();
-            $pfTaskjoblog   = new PluginFusioninventoryTaskjoblog();
+
             $a_agent = $pfAgent->InfosByKey(Toolbox::addslashes_deep($_GET['machineid']));
             if (isset($a_agent['id'])) {
                $moduleRun = $pfTaskjobstate->getTaskjobsAgent($a_agent['id']);
-
                foreach ($moduleRun as $className => $array) {
                   if (class_exists($className)) {
                      if ($className == "PluginFusioninventoryCollect") {
-
-                        $class = new PluginFusioninventoryCollect();
+                        $response['jobs'] = array();
                         foreach ($array as $data) {
-                           $out = $class->run($data, $a_agent);
+                           $out = $pfCollect->run($data, $a_agent);
                            if (count($out) > 0) {
-                              $response[] = $out;
+                              $response['jobs'] = array_merge($response['jobs'], $out);
+                              $response['postmethod'] = 'POST';
+                              $response['token'] = Session::getNewCSRFToken();
                            }
-                           $pfTaskjobstate->changeStatus(
```

Comment:
Why remove the status changes, it nice so see the status of task, in this case see server has sent collect data